### PR TITLE
PR for accesing restored pvc with ROX mode.

### PR DESCRIPTION
**Pull request by avd-sagare**
BZ- https://bugzilla.redhat.com/show_bug.cgi?id=2249601.

### CODE REVIEW

File: tests/functional/pv/pvc_snapshot/test_snapshot_restore_with_different_access_mode.py [hunk 1]
```diff
@@ -57,6 +59,7 @@ def test_snapshot_restore_with_different_access_mode(
                 constants.VOLUME_MODE_FILESYSTEM: [
                     constants.ACCESS_MODE_RWX,
                     constants.ACCESS_MODE_RWO,
+                    constants.ACCESS_MODE_ROX,
```
Comment (jilju) [inline tests/functional/pv/pvc_snapshot/test_snapshot_restore_with_different_access_mode.py:62]: Skip this access mode in the check "Verify md5sum" at the end of this testcase.

**Review by jilju — DISMISSED** [review]

File: tests/functional/pv/pvc_snapshot/test_snapshot_restore_with_different_access_mode.py (hunk 1 above)
Comment (avd-sagare) [reply tests/functional/pv/pvc_snapshot/test_snapshot_restore_with_different_access_mode.py:62]: Done

**Review by jilju — DISMISSED** [review]

**Review by paraggit — DISMISSED** [review]

File: tests/functional/pv/pvc_snapshot/test_snapshot_restore_with_different_access_mode.py [hunk 2]
```diff
@@ -22,6 +23,7 @@
 @skipif_ocs_version("<4.6")
 @skipif_ocp_version("<4.6")
 @polarion_id("OCS-2410")
+@bugzilla("2249601")
```
Comment (PrasadDesala) [inline tests/functional/pv/pvc_snapshot/test_snapshot_restore_with_different_access_mode.py:26]: The main issue in the BZ is the presence of `pool: <>` parameter in cephfs SC and use of ROX clone.
Are we using this pool param in the cephfs SC? and I hope the parent pvc is of type RWX

Comment (avd-sagare) [reply tests/functional/pv/pvc_snapshot/test_snapshot_restore_with_different_access_mode.py:26]: It is present in cephfs SC.
yaml from external storagecluster.

allowVolumeExpansion: true
apiVersion: [storage.k8s.io/v1](http://storage.k8s.io/v1)
kind: StorageClass
metadata:
  annotations:
    description: Provides RWO and RWX Filesystem volumes
  creationTimestamp: "2023-12-19T11:26:35Z"
  name: ocs-external-storagecluster-cephfs
  resourceVersion: "3870482"
  uid: 383b3a6f-6404-472f-9a1c-2c3eb29ec274
parameters:
  clusterID: openshift-storage-extended
  [csi.storage.k8s.io/controller-expand-secret-name](http://csi.storage.k8s.io/controller-expand-secret-name): rook-csi-cephfs-provisioner
  [csi.storage.k8s.io/controller-expand-secret-namespace](http://csi.storage.k8s.io/controller-expand-secret-namespace): openshift-storage-extended
  [csi.storage.k8s.io/node-stage-secret-name](http://csi.storage.k8s.io/node-stage-secret-name): rook-csi-cephfs-node
  [csi.storage.k8s.io/node-stage-secret-namespace](http://csi.storage.k8s.io/node-stage-secret-namespace): openshift-storage-extended
  [csi.storage.k8s.io/provisioner-secret-name](http://csi.storage.k8s.io/provisioner-secret-name): rook-csi-cephfs-provisioner
  [csi.storage.k8s.io/provisioner-secret-namespace](http://csi.storage.k8s.io/provisioner-secret-namespace): openshift-storage-extended
  fsName: fsvol001
  pool: cephfs.fsvol001.data
provisioner: [openshift-storage.cephfs.csi.ceph.com](http://openshift-storage.cephfs.csi.ceph.com/)
reclaimPolicy: Delete
volumeBindingMode: Immediate

**Review by PrasadDesala — APPROVED** [review]

File: tests/functional/pv/pvc_snapshot/test_snapshot_restore_with_different_access_mode.py [hunk 3]
```diff
@@ -182,6 +185,7 @@ def test_snapshot_restore_with_different_access_mode(
                     node_name=next(nodes_iter),
                     pod_dict_path=pod_dict_path,
                     raw_block_pv=pvc_obj.data["spec"]["volumeMode"] == "Block",
+                    pvc_read_only_mode=True,
```
Comment (jilju) [inline tests/functional/pv/pvc_snapshot/test_snapshot_restore_with_different_access_mode.py:188]: The value be True only when the access mode is ROX.

**Review by jilju — APPROVED** [review]

**Review by PrasadDesala — APPROVED** [review]

**Pull request merged**